aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/EventSourceType.java

```diff
@@ -0,0 +1,194 @@
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.mobileconnectors.pinpoint.targeting.notification;
+
+import android.os.Bundle;
+import android.util.Log;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSyntaxException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.CAMPAIGN_ACTIVITY_ID_ATTRIBUTE_KEY;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.CAMPAIGN_ID_ATTRIBUTE_KEY;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.CAMPAIGN_PUSH_KEY_PREFIX;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.JOURNEY_ACTIVITY_ID_ATTRIBUTE_KEY;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.JOURNEY_ID_ATTRIBUTE_KEY;
+
+public class EventSourceType {
+    private static final String TAG = EventSourceType.class.getSimpleName();
+    private static final String AWS_EVENT_TYPE_OPENED = "opened_notification";
+    private static final String AWS_EVENT_TYPE_RECEIVED_FOREGROUND = "received_foreground";
+    private static final String AWS_EVENT_TYPE_RECEIVED_BACKGROUND = "received_background";
+    private static final String PINPOINT_ATTRIBUTE_KEY = "pinpoint";
+    private static final String JOURNEY_ATTRIBUTE_KEY = "journey";
+    private final EventSourceAttributeParser attributeParser;
+    private String eventSourceName;
+    private String eventTypeOpenend;
+    private String eventTypeReceivedForeground;
+    private String eventTypeReceivedBackground;
+    private String eventSourceIdAttributeKey;
+    private String eventSourceActivityIdAttributeKey;
+    private String eventSourceKeyPrefix;
+
+    /**
+     * Campaign attributes are send from pinpoint flattened
+     * For example:
+     *  "pinpoint.campaign.campaign_id"
+     * Journey attributes come in JSON format
+     * This interface just seeks to abstract some of that away
+     * from the logic that handles the notfication
+     * @return
+     */
+    public interface EventSourceAttributeParser {
+        Map<String, String> getEventSourceAttributes(Bundle bundle);
+    }
+
+    private static final EventSourceAttributeParser CAMPAIGN_PARSER = new EventSourceAttributeParser() {
+        @Override
+        public Map<String, String> getEventSourceAttributes(Bundle bundle) {
+            if(bundle == null) {
+                return null;
```

Comment:
Prefer getting out of `null` as quickly as you can. Prefer return of `@NonNull`, `Collections.emptyMap()`.

Ref a discussion https://stackoverflow.com/q/45009329

(Apply same default return to all location where you return a `null` collection in this PR.)

Or, since your methods are mostly starting with a `new HashMap<>()` and the populating values into it, you could return _that_ object, as your empty collection, instead of the (explicitly stated) `Collections.emptyMap()`.

Reply:
It looks like the `@NonNull` annotation comes from `com.android.support:support-annotations` dependency. Looking through the SDK, doesn't look like any of the modules take a dependency on that right now. Is that something we can add?